foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_8_0/OrphedomosToFabric8Migration.java

```diff
@@ -40,237 +39,374 @@
-     */
-    private Plugin getPlugin(Model model, String pluginName) {
-        Build build = model.getBuild();
-        if (build != null) {
-            List<Plugin> plugins = build.getPlugins();
-            for (Plugin plugin : plugins) {
-                if ((pluginName).equalsIgnoreCase(plugin.getArtifactId())) {
-                    return plugin;
-                }
-            }
-        }
-        return null;
-    }
-
-    private boolean isRootProjectPom(Model model) {
-        return ROOT_GROUP_ID.equals(model.getParent().getGroupId()) && ROOT_ARTIFACT_ID.equals(model.getParent().getArtifactId());
-    }
-    private boolean isTestDockerPom(Model model){
-        return model.getParent().getArtifactId().endsWith(TESTS_DOCKER_PARENT_ARTIFACT_ID) && model.getArtifactId().endsWith(TESTS_DOCKER_ARTIFACT_ID);
-    }
-
-    private Xpp3Dom getConfiguration(Plugin plugin) {
-        Object pluginConfiguration = plugin.getConfiguration();
-        return (Xpp3Dom) pluginConfiguration;
-    }
-
-    private static boolean hasConfigurationItem(Xpp3Dom configuration, String configurationItem){
-        return configuration.getChildren(configurationItem).length > 0;
-    }
+    private static String repoUrl;
+    private static String profile;
 
     /**
      * Function to validate whether migration should be performed or not.
+     *
      * @param file file to validate
      * @return true if migration should be performed
      */
     @Override
     protected boolean shouldExecuteOnFile(File file) {
-        boolean shouldExecute = false;
         boolean migrateConfiguration = false;
+        boolean migrateProfile = false;
         boolean migratePackaging = false;
 
         Model model = getLocationAnnotatedModel(file);
-        if (isRootProjectPom(model) || isTestDockerPom(model)){
-            shouldExecute = false;
-        } else {
-            Plugin orphedomosPlugin = getPlugin(model, ORPHEDOMOS_ARTIFACT_ID);
-            if (orphedomosPlugin != null){
-                migrateConfiguration = true;
-            }
-            if(containsOrphedomosPackaging(file)){
-                migratePackaging = true;
-            }
-            shouldExecute = migrateConfiguration || migratePackaging;
+
+        Plugin orphedomosBuildPlugin = getBuildPlugin(model, ORPHEDOMOS_ARTIFACT_ID);
+        if (orphedomosBuildPlugin != null) {
+            migrateConfiguration = true;
         }
 
-        return shouldExecute;
+        Plugin orphedomosCiProfilePlugin = getProfilePlugin(model, ORPHEDOMOS_ARTIFACT_ID, CI);
+        Plugin orphedomosItProfilePlugin = getProfilePlugin(model, ORPHEDOMOS_ARTIFACT_ID, INTEGRATION_TEST);
+
+        if (orphedomosCiProfilePlugin != null || orphedomosItProfilePlugin != null) {
+            migrateProfile = true;
+        }
+
+        if (containsOrphedomosPackaging(file)) {
+            migratePackaging = true;
+        }
+
+        return migrateConfiguration || migrateProfile || migratePackaging;
```

Comment:
Could simplify to `return orphedomosBuildPlugin != null || orphedomosCiProfilePlugin != null || orphedomosItProfilePlugin != null || containsOrphedomosPackaging(file)`